proof/refine/ARM/KHeap_R.thy

```diff
@@ -3917,6 +3914,180 @@ lemma refillSufficient_wp:
   apply (wpsimp wp: getRefills_wp)
   by (clarsimp simp: sufficientRefills_def obj_at'_def)
 
+(* projection rewrites *)
+
+abbreviation
+  opt_pred :: "('a \<Rightarrow> bool) \<Rightarrow> ('b \<Rightarrow> 'a option) \<Rightarrow> ('b \<Rightarrow> bool)" (infixl "|<" 50) where
```

Comment:
Could you please move opt-pred to the option mane file (roughly where opt-map is defined)